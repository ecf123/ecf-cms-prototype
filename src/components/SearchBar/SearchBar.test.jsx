import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import SearchBar from "../SearchBar/SearchBar.jsx";

import { customRender } from "../../utils/testUtils";

it("Should render searchbar with children", () => {
  const { container } = customRender(
    <SearchBar>
      <p>some content</p>
    </SearchBar>
  );
  expect(container).toMatchSnapshot();
});


it("Should update input box with each key press", () => {
  render(<SearchBar />)

  const input = screen.getByTestId("search-input");
  userEvent.type(input, "financial forrest")

  const updatedInput = screen.findByText("financial forrest");
  expect(updatedInput).toBeTruthy();
});


it("Should not display the wrong order of characters based on what was typed", () => {
    render(<SearchBar />)
  
    const input = screen.getByTestId("search-input");
    userEvent.type(input, "financial forrest")
  
    const updatedInput = screen.queryByText("forrest financial");
    expect(updatedInput).toBeFalsy();
  });

  it("Should render in the search bubble", () => {
    render(<SearchBar />)
  
    const input = screen.getByTestId("search-input");
  
    expect(input).toBeInTheDocument();
  });