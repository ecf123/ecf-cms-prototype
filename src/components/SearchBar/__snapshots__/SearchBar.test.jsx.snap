// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render searchbar with children 1`] = `
<div>
  <div
    class="search-bar"
  >
    <img
      alt="Search Icon"
      class="search-bar__image"
      src="magnifying-glass-grey.svg"
    />
    <input
      class="search-bar__input"
      data-testid="search-input"
      type="text"
      value=""
    />
  </div>
</div>
`;
